torch/_refs/__init__.py

```diff
@@ -3823,7 +3831,7 @@ def _reshape_view_helper(a: TensorLikeType, *shape, allow_copy: bool) -> TensorL
             a_ = flatten(a_, idx, end)
 
         # Splits the (possibly flattened) dimension to create the desired dim length
-        if guard_size_oblivious(accum != length):
+        if guard_or_true(accum != length):
```

Comment:
We did a test to verify that this check is ok to by pass, the squeeze bellow will offset an additional splitting here. 
https://github.com/pytorch/pytorch/pull/150584